src/functTest/groovy/com/bmuschko/gradle/docker/tasks/container/DockerCopyFileToContainerFunctionalTest.groovy

```diff
@@ -68,7 +69,7 @@ class DockerCopyFileToContainerFunctionalTest extends AbstractGroovyDslFunctiona
                 dependsOn createContainer, createTarFile
                 finalizedBy removeContainer
                 targetContainerId createContainer.getContainerId()
-                tarFile = new File("$projectDir/HelloWorld.tgz")
+                tarFile = new File("${TestUtils.escapeFilePath(projectDir)}/HelloWorld.tgz")
```

Comment:
We'll likely want `"TestUtils.escapeFilePath(new File(projectDir, 'HelloWorld.tgz')"` here. Was there any reason you only escaped the directory but not the rest. Same thing for some other occurrences below.

Reply:
Not entirely sure I understand here. The resulting text line we need would be: -

tarFile = new File("c:\\testdir/HelloWorld.tgz")
or perhaps better
tarFile = new File("c:\\testdir", 'HelloWorld.tgz')

We only need to escape the backslashes in projectDir, not the forward slash (allowed + interpreted correctly)

escapeFilePath returns a string so you would need: -
tarFile = new File("${TestUtils.escapeFilePath(new File(projectDir, 'HelloWorld.tgz')}"

but I don't think you gain anything as you only need to escape the projectDir. The nicer code I think is: -
tarFile = new File("${TextUtils.escapeFilePath(projectDir)}", 'HelloWorld.tgz')

I'll update these in the review and you'll have to clarify if I am missing something.

Reply:
Think I understand
"TestUtils.escapeFilePath(new File(projectDir, 'HelloWorld.tgz')" is nicer everywhere else, where we are not creating a file